bundles/org.smarthomej.binding.amazonechocontrol/src/main/java/org/smarthomej/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -785,12 +786,35 @@ private void handlePushActivity(@Nullable String payload) {
         }
 
         String search = key.registeredUserId + "#" + key.entryId;
-        connection.getActivities(10, pushActivity.timestamp).stream().filter(activity -> search.equals(activity.id))
-                .findFirst()
-                .ifPresent(currentActivity -> currentActivity.getSourceDeviceIds().stream()
-                        .map(sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
-                        .filter(Objects::nonNull).forEach(echoHandler -> Objects.requireNonNull(echoHandler)
-                                .handlePushActivity(currentActivity)));
+
+        /*
+         * DEPRECATED
+         * connection.getActivities(10, pushActivity.timestamp).stream().filter(activity -> search.equals(activity.id))
+         * .findFirst()
+         * .ifPresent(currentActivity -> currentActivity.getSourceDeviceIds().stream()
+         * .map(sourceDeviceId -> findEchoHandlerBySerialNumber(sourceDeviceId.serialNumber))
+         * .filter(Objects::nonNull).forEach(echoHandler -> Objects.requireNonNull(echoHandler)
+         * .handlePushActivity(currentActivity)));
+         */
+
+        Long timestamp = pushActivity.timestamp;
+        if (timestamp != null) {
+            Long startTimestamp = timestamp - 30000;
+            Long endTimestamp = timestamp + 30000;
+            List<CustomerHistoryRecord> customerHistoryRecords = connection.getActivities(startTimestamp, endTimestamp);
+            for (CustomerHistoryRecord customerHistoryRecord : customerHistoryRecords) {
+                if (customerHistoryRecord != null) {
+                    String recordKey = customerHistoryRecord.recordKey;
+                    if (recordKey != null && search.equals(recordKey)) {
+                        EchoHandler echoHandler = findEchoHandlerBySerialNumber(recordKey.split("#")[3]);
```

Comment:
Is this safe? If recordKey.split returns less than four parts, this will result in an IndexOutOfBoundsException

Reply:
It depends on the data of the response, so i will add a check.